sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceImpl.java

```diff
@@ -315,29 +319,36 @@ protected String buildGetRowsByPKCQL(
       String keyspaceName,
       String tableName,
       List<PathSegment> pkValues,
-      List<String> columns,
+      List<Column> columns,
       Schema.CqlTable tableDef,
       QueryOuterClass.Values.Builder valuesBuilder,
       ToProtoConverter toProtoConverter) {
-    SelectFrom selectFrom = QueryBuilder.selectFrom(keyspaceName, tableName);
-    Select select;
-    if (columns.isEmpty()) {
-      select = selectFrom.all();
-    } else {
-      select = selectFrom.columns(columns);
-    }
     final int keysIncluded = pkValues.size();
     final List<QueryOuterClass.ColumnSpec> primaryKeys =
         getAndValidatePrimaryKeys(tableDef, keysIncluded);
+    List<BuiltCondition> whereConditions = new ArrayList<>();
```

Comment:
Since the size is known what do you thing about setting the initial capacity? 
```
List<BuiltCondition> whereConditions = new ArrayList<>(keysIncluded);
```

Reply:
I can do that sure.